x-pack/metricbeat/module/azure/app_insights/data.go

```diff
@@ -66,7 +103,9 @@ func getMetric(addProp map[string]interface{}) map[string]interface{} {
 		switch val.(type) {
 		case map[string]interface{}:
 			for subKey, subVal := range val.(map[string]interface{}) {
-				metricNames[cleanMetricNames(fmt.Sprintf("%s.%s", key, subKey))] = subVal
+				if subVal != nil {
```

Comment:
Would the removal of nil values need another changelog entry?

Reply:
I do not think so, this is normal behavior, the app insights have no data for these metrics so no data will be sent.